claims.go

```diff
@@ -400,33 +315,46 @@ func (p *Claims) validateHwVersion() error {
 	return nil
 }
 
-func (p *Claims) validateNonce() error {
-	if p.Nonce == nil {
-		return fmt.Errorf("missing mandatory nonce")
+func (c *Claims) validateNonce(expectedProfile string) error {
+	var nonce []byte
+	switch expectedProfile {
+	case PSA_PROFILE_1:
+		if c.LegacyNonce == nil {
+			return fmt.Errorf("missing mandatory nonce")
+		}
+		nonce = *c.LegacyNonce
+	case PSA_PROFILE_2:
+		if c.Nonce == nil {
+			return fmt.Errorf("missing mandatory nonce")
+		}
+		if c.Nonce.Len() != 1 {
+			return errors.New("there must be exactly one nonce")
+		}
+		nonce = c.Nonce.GetI(0)
```

Comment:
Here also, would prefer to see a const like FirstIndex

Reply:
not sure this would add more clarity